src/main/java/com/shopify/store/Application.java

```diff
@@ -0,0 +1,57 @@
+package com.shopify.store;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+
+/**
+ *  Application here where one can enter and add in product, we can also input an ENUM check in the future to look at categories of products
+ */
+public class Application {
+    public static BigDecimal cart (ArrayList<FruitProduct> cartList) {
+        var discountZero = new Discount(new DiscountZeroStrategy());
+        var discountBOGO = new Discount(new DiscountBOGOStrategy());
+        var discountPercent = new Discount(new DiscountPercentageStrategy());
+       BigDecimal total = new BigDecimal(0);
+       for (int i = 0; i < cartList.size(); i++) {
+           FruitProduct product = cartList.get(i);
+           if(product.getProductName().equalsIgnoreCase("apples") && product.getProductQuantity() >= 2) {
+                total = total.add(discountPercent.applyDiscount(product.getPrice(), product.getProductQuantity(), 0.2));
+           } else if(product.getProductName().equalsIgnoreCase("grapes") && product.getProductQuantity() >= 2) {
+               total = total.add(discountBOGO.applyDiscount(product.getPrice(), product.getProductQuantity(),0.5));
+           }
+           else {
+               total = total.add(discountZero.applyDiscount(product.getPrice(), product.getProductQuantity(),1));
+           }
+       }
+       return total;
+    }
```

Comment:
This code designed doesn't scale. Imagine you had a catalog of 1000 items, where 100 of them have some kind of promotion attached. Imagine if this `if`/`else if`/`else` ladder was 100 items tall.

Reply:
Yeah, I think i can move all of this into the Concrete Strategy side of things and eliminate this, i'll try that thanks